blocks/store/selectors.js

```diff
@@ -25,9 +27,32 @@ export const getBlockTypes = createSelector(
  *
  * @return {Object?} Block Type.
  */
-export function getBlockType( state, name ) {
-	return state.blockTypes[ name ];
-}
+export const getBlockType = createSelector(
+	( state, name ) => {
+		const blockTypeDefinition = state.blockTypes[ name ];
+		const blockTypeImplementation = state.implementations[ name ];
+
+		if ( ! blockTypeDefinition || ! blockTypeImplementation ) {
+			return null;
+		}
+
+		return {
+			...blockTypeDefinition,
+			...blockTypeImplementation,
```

Comment:
It makes sense to me that the implementation can override attributes that are in the definition. 

> What about the icon? It's considered as implementation for now mainly because we allow svg elements which can't be defined server-side. An option would be to transform it to svg string to allow it to be moved to the definition of the block.

We could have `icon` supported both in `addBlockType` and `implementBlockType`. SVGs would only be supported in `implementBlockType`. The attribute specified in the implementation takes priority, as above.

Reply:
>It makes sense to me that the implementation can override attributes that are in the definition.

It's not quite as obvious to me. What's the use case?